### PR TITLE
remove python 2 support

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- removed python 2 compatible code
- reduced build to run on py3.x (travis)
- won't build a python 2 wheel no more
- removed `six` dependency

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [x] updated the changelog

### CODE REVIEW

File: rasa_nlu/data_router.py [hunk 1]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/data_router.py]:
```suggestion
```

File: rasa_nlu/emulators/luis.py [hunk 2]
```diff
@@ -1,7 +1,3 @@
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import division
-from __future__ import absolute_import
```
Comment (ricwo) [inline rasa_nlu/emulators/luis.py]:
```suggestion
```

File: rasa_nlu/emulators/wit.py [hunk 3]
```diff
@@ -1,7 +1,3 @@
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import division
-from __future__ import absolute_import
```
Comment (ricwo) [inline rasa_nlu/emulators/wit.py]:
```suggestion
```

File: rasa_nlu/evaluate.py [hunk 4]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/evaluate.py]:
```suggestion
```

File: rasa_nlu/extractors/__init__.py [hunk 5]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/extractors/__init__.py]:
```suggestion
```

File: rasa_nlu/extractors/crf_entity_extractor.py [hunk 6]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/extractors/crf_entity_extractor.py]:
```suggestion
```

File: rasa_nlu/extractors/duckling_http_extractor.py [hunk 7]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/extractors/duckling_http_extractor.py]:
```suggestion
```

File: rasa_nlu/extractors/entity_synonyms.py [hunk 8]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/extractors/entity_synonyms.py]:
```suggestion
```

File: rasa_nlu/extractors/mitie_entity_extractor.py [hunk 9]
```diff
@@ -1,7 +1,3 @@
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import division
-from __future__ import absolute_import
```
Comment (ricwo) [inline rasa_nlu/extractors/mitie_entity_extractor.py]:
```suggestion
```

File: rasa_nlu/extractors/spacy_entity_extractor.py [hunk 10]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/extractors/spacy_entity_extractor.py]:
```suggestion
```

File: rasa_nlu/featurizers/__init__.py [hunk 11]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/featurizers/__init__.py]:
```suggestion
```

File: rasa_nlu/featurizers/count_vectors_featurizer.py [hunk 12]
```diff
@@ -1,7 +1,3 @@
-from __future__ import division
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import absolute_import
```
Comment (ricwo) [inline rasa_nlu/featurizers/count_vectors_featurizer.py]:
```suggestion
```

File: rasa_nlu/featurizers/mitie_featurizer.py [hunk 13]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/featurizers/mitie_featurizer.py]:
```suggestion
```

File: rasa_nlu/featurizers/ngram_featurizer.py [hunk 14]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/featurizers/ngram_featurizer.py]:
```suggestion
```

File: rasa_nlu/featurizers/regex_featurizer.py [hunk 15]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/featurizers/regex_featurizer.py]:
```suggestion
```

File: rasa_nlu/featurizers/spacy_featurizer.py [hunk 16]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/featurizers/spacy_featurizer.py]:
```suggestion
```

File: rasa_nlu/model.py [hunk 17]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/model.py]:
```suggestion
```

File: rasa_nlu/persistor.py [hunk 18]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/persistor.py]:
```suggestion
```

File: rasa_nlu/run.py [hunk 19]
```diff
@@ -1,12 +1,7 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/run.py]:
```suggestion
```

File: rasa_nlu/server.py [hunk 20]
```diff
@@ -1,14 +1,9 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/server.py]:
```suggestion
```

File: rasa_nlu/tokenizers/jieba_tokenizer.py [hunk 21]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/tokenizers/jieba_tokenizer.py]:
```suggestion
```

File: rasa_nlu/tokenizers/spacy_tokenizer.py [hunk 22]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/tokenizers/spacy_tokenizer.py]:
```suggestion
```

File: rasa_nlu/tokenizers/whitespace_tokenizer.py [hunk 23]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/tokenizers/whitespace_tokenizer.py]:
```suggestion
```

File: rasa_nlu/train.py [hunk 24]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/train.py]:
```suggestion
```

File: rasa_nlu/training_data/__init__.py [hunk 25]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/__init__.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/__init__.py [hunk 26]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/__init__.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/dialogflow.py [hunk 27]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/dialogflow.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/luis.py [hunk 28]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/luis.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/markdown.py [hunk 29]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/markdown.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/rasa.py [hunk 30]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/rasa.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/readerwriter.py [hunk 31]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/readerwriter.py]:
```suggestion
```

File: rasa_nlu/training_data/formats/wit.py [hunk 32]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/formats/wit.py]:
```suggestion
```

File: rasa_nlu/training_data/loading.py [hunk 33]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/loading.py]:
```suggestion
```

File: rasa_nlu/training_data/message.py [hunk 34]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/message.py]:
```suggestion
```

File: rasa_nlu/training_data/training_data.py [hunk 35]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/training_data.py]:
```suggestion
```

File: rasa_nlu/training_data/util.py [hunk 36]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/training_data/util.py]:
```suggestion
```

File: rasa_nlu/utils/__init__.py [hunk 37]
```diff
@@ -1,9 +1,6 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/utils/__init__.py]:
```suggestion
```

File: rasa_nlu/utils/mitie_utils.py [hunk 38]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/utils/mitie_utils.py]:
```suggestion
```

File: rasa_nlu/utils/spacy_utils.py [hunk 39]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline rasa_nlu/utils/spacy_utils.py]:
```suggestion
```

File: rasa_nlu/version.py [hunk 40]
```diff
@@ -1,6 +1,2 @@
-from __future__ import unicode_literals
-from __future__ import print_function
-from __future__ import division
-from __future__ import absolute_import
```
Comment (ricwo) [inline rasa_nlu/version.py]:
```suggestion
```

File: tests/base/test_components.py [hunk 41]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_components.py]:
```suggestion
```

File: tests/base/test_config.py [hunk 42]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_config.py]:
```suggestion
```

File: tests/base/test_data_router.py [hunk 43]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_data_router.py]:
```suggestion
```

File: tests/base/test_emulators.py [hunk 44]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_emulators.py]:
```suggestion
```

File: tests/base/test_emulators.py [hunk 45]
```diff
@@ -1,7 +1,3 @@
...
```
Comment (ricwo) [inline tests/base/test_emulators.py]:
```suggestion
```

File: tests/base/test_featurizers.py [hunk 46]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_featurizers.py]:
```suggestion
```

File: tests/base/test_interpreter.py [hunk 47]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_interpreter.py]:
```suggestion
```

File: tests/base/test_multitenancy.py [hunk 48]
```diff
@@ -1,8 +1,4 @@
 # -*- coding: utf-8 -*-
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_multitenancy.py]:
```suggestion
```

File: tests/base/test_persistor.py [hunk 49]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_persistor.py]:
```suggestion
```

File: tests/base/test_project.py [hunk 50]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_project.py]:
```suggestion
```

File: tests/base/test_server.py [hunk 51]
```diff
@@ -1,8 +1,4 @@
 # -*- coding: utf-8 -*-
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_server.py]:
```suggestion
```

File: tests/base/test_synonyms.py [hunk 52]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/base/test_synonyms.py]:
```suggestion
```

File: tests/conftest.py [hunk 53]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/conftest.py]:
```suggestion
```

File: tests/example_component.py [hunk 54]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/example_component.py]:
```suggestion
```

File: tests/training/test_train.py [hunk 55]
```diff
@@ -1,9 +1,5 @@
 # -*- coding: utf-8 -*-
 
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/training/test_train.py]:
```suggestion
```

File: tests/utilities.py [hunk 56]
```diff
@@ -1,7 +1,3 @@
-from __future__ import absolute_import
-from __future__ import division
-from __future__ import print_function
-from __future__ import unicode_literals
```
Comment (ricwo) [inline tests/utilities.py]:
```suggestion
```

**Comment by tmbo** [comment]
took me 20 suggestions to find the button where I can apply them in a batch :D

**Comment by ricwo** [comment]
😄 I was just about to send you [this](https://help.github.com/articles/incorporating-feedback-in-your-pull-request/) 

wait, did you see my comments - https://github.com/RasaHQ/rasa_nlu/pull/1641#pullrequestreview-195633349?

**Comment by ricwo** [comment]
also, `from future.utils import PY3` in `sklearn_intent_classifier.py` (was part of the review so I didn't list it among the comments above)

**Comment by tmbo** [comment]
the future import will be fixed in another PR but the dockerfile import is good, will fix that on master

**Pull request merged**